hub/api/dataset.py

```diff
@@ -94,6 +95,31 @@ def __new__(
         except AgreementError as e:
             raise e from None
 
+    @staticmethod
+    def exists(path: str) -> bool:
```

Comment:
let's add creds and token as argument to this as well

Reply:
Okay, will do!